eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -17,46 +11,30 @@ steps:
       Write-Host "##vso[task.prependpath]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES_SECRET;issecret=true;]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
     displayName: Set Vcpkg Variables
-    condition: >-
-      and(
-        succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
-      )
 
-  # This task times out after 10 minutes. An analysis of a subset of Vcpkg Cache
-  # runs showed that after 10 minutes the cache step is very unlikely to
-  # succeed.
-  - task: Cache@2
-    inputs:
-      key: >-
-        $(Agent.JobName)
-        | "$(VcpkgCommit)"
-        | $(Agent.Os)
-        | $(${{ parameters.DependenciesVariableName }})
-      path: $(VCPKG_INSTALLATION_ROOT)
-      cacheHitVar: VcpkgRestoredFromCache
-    displayName: Vcpkg Cache
-    timeoutInMinutes: 10
+  - pwsh: |
+      if ([string]::IsNullOrEmpty($env:VCPKG_CACHE_SAS)) {
+        throw 'No VCPKG SAS'
+      }
+      $SasOnly=$env:VCPKG_CACHE_SAS
+      if ($env:VCPKG_CACHE_SAS.ToCharArray()[0] -eq "?") {
```

Comment:
@vhvb1989 why do we need to do this here? Are you sure the SAS token has the correct value? I'd suggest you do any such transform inside the KV so you don't need to do any translations here.

Reply:
Well, I was learning how to make this work during a Friday night w/o EngSys support. And I learned:

- When a variable is set as a secret, it needs to be mapped to any step/job which will need it.
- The SAS value (coming from EngSys contains a `?` as the first character. Probably copied from Azure Portal). And that's not useful here. So I needed to remove that character before using the value.
- After removing the character, the SAS has not write-access, so the `upload` cache operation failed.

At that point I gave up and added a comment about I need EngSys to provide a new SAS with write access.  And if the new SAS is provided without the `?` char at the start, then I can remove this logic to remove it.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3439